src/sql/azdata.d.ts

```diff
@@ -2764,6 +2764,7 @@ declare module 'azdata' {
 		withValidation(validation: (component: TComponent) => boolean | Thenable<boolean>): ComponentBuilder<TComponent, TPropertyBag>;
 	}
 	export interface ContainerBuilder<TComponent extends Component, TLayout, TItemLayout, TPropertyBag extends ComponentProperties> extends ComponentBuilder<TComponent, TPropertyBag> {
+		withProps(properties: TPropertyBag): ComponentBuilder<TComponent, TPropertyBag>;
```

Comment:
Why are you adding this one here?

Reply:
So that we can do flexContainer.withProps().withLayout() as withprops() was returning the base class which did not support withLayout() method.

Reply:
Yes, but why is it here instead of proposed?